datasets/WiderFace_to_tfrecords.py

```diff
@@ -0,0 +1,241 @@
+      shape: 3 integers, image shapes in pixels.
+    Returns:
+      Example proto
+    """
+    xmin = []
+    ymin = []
+    xmax = []
+    ymax = []
+    shape = []
+    difficult = []
+    truncated = []
+    label_text =[]
+    for i in range(len(data_example['bboxes'])):
+        difficult.append(0)
+        truncated.append(0)
+        label_text.append(b'face')
+
+    for s in data_example['shape']:
+        shape.append(s)
+    # print(shape)
+    image_data = data_example['image']
+    filename = data_example['name']
+    for bbox in data_example['bboxes']:
+        assert len(bbox) == 4
+        # pylint: disable=expression-not-assigned
+        xmin.append(bbox['xmin'] / shape[1])
+        ymin.append(bbox['ymin'] / shape[0])
+        xmax.append(bbox['xmax'] / shape[1])
+        ymax.append(bbox['ymax'] / shape[0])
+        # pylint: enable=expression-not-assigned
+
+    # print(xmin)
+    image_format = b'JPEG'
+    example = tf.train.Example(features=tf.train.Features(feature={
+        'image/height': int64_feature(shape[0]),
+        'image/width': int64_feature(shape[1]),
+        'image/channels': int64_feature(shape[2]),
+        'image/shape': int64_feature(shape),
+        'image/object/bbox/xmin': float_feature(xmin),
+        'image/object/bbox/xmax': float_feature(xmax),
+        'image/object/bbox/ymin': float_feature(ymin),
+        'image/object/bbox/ymax': float_feature(ymax),
+        'image/object/bbox/label': int64_feature(data_example['labels']),
+        'image/object/bbox/label_text': bytes_feature(label_text),
+        'image/object/bbox/difficult': int64_feature(difficult),
+        'image/object/bbox/truncated': int64_feature(truncated),
+        'image/format': bytes_feature(image_format),
+        'image/filename': bytes_feature(filename.encode('utf-8')),
+        'image/encoded': bytes_feature(image_data)}))
+    return example
+
+
+def _add_to_tfrecord(info, dataset_path, tfrecord_writer):
+    """Loads data from image and annotations files and add them to a TFRecord.
+
+    Args:
+      dataset_dir: Dataset directory;
+      name: Image name to add to the TFRecord;
+      tfrecord_writer: The TFRecord writer to use for writing.
+    """
+    data_example = \
+        _process_image(info, dataset_path)
+    example = _convert_to_example(data_example)
+    tfrecord_writer.write(example.SerializeToString())
+
+
+def _get_output_filename(output_dir, name):
+    return '%s/%s.tfrecord' % (output_dir, name)
+
+
+def _get_dataset_filename(output_dir, name, shard_id, num_shard, records_num):
+    output_filename = '%s_%05d-of-%05d-total%05d.tfrecord' % (name, shard_id + 1, num_shard, records_num)
+    return os.path.join(output_dir, output_filename)
+
+
+def run(anno_path, dataset_path, output_dir, name, shuffling=False):
+    # Dataset filenames, and shuffling.
+
+    if not os.path.isdir(output_dir):
+        os.mkdir(output_dir)
+    annotations = get_annotations(anno_path)
+
+    if shuffling:
```

Comment:
I think it can't shuffle the annotation simply.
The structure of the content will be lost.

Reply:
I make a mistake, the annotation is not the original widerface annotation file